/*
    Licensed under the Apache License, Version 2.0 (the "License");
    you may not use this file except in compliance with the License.
    You may obtain a copy of the License at

        https://www.apache.org/licenses/LICENSE-2.0

    Unless required by applicable law or agreed to in writing, software
    distributed under the License is distributed on an "AS IS" BASIS,
    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
    See the License for the specific language governing permissions and
    limitations under the License.
*/

syntax = "proto3";
import "google/protobuf/any.proto";
package NES;

option java_package = "stream.nebula.protobuf";
option java_multiple_files = true;

message SerializableShapeType {
  // Enum to identify the shape type
  enum ShapeType {
    UNDEFINED = 0;
    CIRCLE = 1;
    POINT = 2;
    POLYGON = 3;
    RECTANGLE = 4;
  }

  message SerializablePoint {
    double latitude = 1;
    double longitude = 2;
  }

  message SerializableCircle {
    double latitude = 1;
    double longitude = 2;
    double radius = 3;
  }

  message SerializablePolygon {
    repeated double coords = 1;
  }

  message SerializableRectangle {
    double latitudeLow = 1;
    double longitudeLow = 2;
    double latitudeHigh = 3;
    double longitudeHigh = 4;
  }

  ShapeType shapeType = 1;
  google.protobuf.Any details = 2;
}


